bouvet-deler/databricks_lambda/databricks_lambda_case2.py

```diff
@@ -0,0 +1,76 @@
+# Databricks notebook source
+# MAGIC %md
+# MAGIC # Using Lambda in Databricks - Case 2
+# MAGIC 
+# MAGIC This notebooks gives a slightly complex example of using lambda in Databricks. For more information see this article (add link).
+# MAGIC 
+# MAGIC Author: Morten Mjelde <morten.mjelde@bouvet.no>
+
+# COMMAND ----------
+
+# Define DataSet3
+from pyspark.sql import Row
+
+DataSet3 = spark.createDataFrame(
+    [
+        Row(location="Oslo", people=[Row(firstName="Harald", lastName="Haraldsen"), Row(firstName="Mark", lastName="Hunter")]),
+        Row(location="Stockholm", people=[Row(firstName="Ingrid", lastName="Hansen")]),
+    ]
+)
+
+# COMMAND ----------
+
+# Define DataSet4
+from pyspark.sql import Row
+
+DataSet4 = spark.createDataFrame(
+    [
+        Row(location="Copenhagen", people=[Row(firstName="Roger", nobleName="of house Henriksen"), Row(firstName="Hans", nobleName="of house Dale")]),
+    ]
+)
+
+# COMMAND ----------
+
+# Define transformations
+from pyspark.sql.functions import concat, col, DataFrame, lit, explode
+
+def explodePeople(df: DataFrame) -> DataFrame:
+    """Explode people column."""
+    return df.withColumn("ex_people", explode("people")).select("location", "ex_people.*")
+
+def addInitials(df: DataFrame) -> DataFrame:
+    """Add initials column."""
+    df_new = df.withColumn("intials", concat(col("firstName").substr(0,1), col("lastName").substr(0,1)))
+    return df_new
+
+def addfullName(df: DataFrame) -> DataFrame:
+    """Add full name column."""
+    return df.withColumn("fullName", concat(col("firstName"), lit(" "), col("lastName")))
+
+def nameTransformation(data: DataFrame, correctNameColumns: callable = lambda df: df) -> DataFrame:
+    """Perform all transformations."""
+    data = explodePeople(data)
```

Comment:
have to love the naming here 🤣 and the column over should then be `withColumn("exploded_people",......)`